pkg/config/node.go

```diff
@@ -519,30 +520,32 @@ func getNodeConfig(kubeconfigPath, clusterConfigPath, resolvConfPath string, api
 			node.Cluster.IngressVIPEmptyType = "A"
 		}
 	}
-	// Rest of the Node config will not be available on Cloud platforms.
-	if onPremPlatform, err := isOnPremPlatform(clusterConfigPath); !onPremPlatform {
+	// Check if running on a cloud platform
+	onPremPlatform, err := isOnPremPlatform(clusterConfigPath)
+	if err != nil {
 		return node, err
 	}
-
-	vipIface, nonVipAddr, err := GetVRRPConfig(apiVip, ingressVip)
+	resolvConfUpstreams, err := getDNSUpstreams(resolvConfPath)
 	if err != nil {
 		return node, err
 	}
-	node.NonVirtualIP = nonVipAddr.IP.String()
 
-	node.EnableUnicast = false
-	if os.Getenv("ENABLE_UNICAST") == "yes" {
-		node.EnableUnicast = true
-	}
+	invalidUpstreams := []string{"127.0.0.1", "::1"}
```

Comment:
What I mean, something like `loopbackIPs := []string{"127.0.0.1", "::1"}` and then `invalidUpstreams = append(loopbackIPs, node.NonVirtualIP)`. Otherwise it feels a bit like a magic and I can expect a new member of the team to ask "why are those 2 IPs invalid?"

Reply:
Go does not support constant arrays, maps or slices. I have incorporated the above 2 comments.

Reply:
Of course, that was stupid suggestion but you got the idea amazingly well, thanks for fixing!